sdk/identity/Azure.Identity/tests/AuthenticationRecordTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace Azure.Identity.Tests
+{
+    public class AuthenticationRecordTests
+    {
+        [Test]
+        public void SerializeDeserializeInputChecks()
+        {
+            var record = new AuthenticationRecord(Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString());
+
+            Assert.Throws<ArgumentNullException>(() => record.Serialize(null));
+            Assert.ThrowsAsync<ArgumentNullException>(async () => await record.SerializeAsync(null));
+            Assert.Throws<ArgumentNullException>(() => AuthenticationRecord.Deserialize(null));
+            Assert.ThrowsAsync<ArgumentNullException>(async () => await AuthenticationRecord.DeserializeAsync(null));
+        }
+
+        [Test]
+        public async Task SerializeDeserializeAsync()
+        {
+            var expRecord = new AuthenticationRecord(Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString());
+
+            byte[] buff = new byte[256];
```

Comment:
Nit - might be nice to pull that into a BufferSize constant so it's easy to scale when you add three more fields to AuthRecord